tests/StacktraceTest.php

```diff
@@ -188,6 +188,34 @@ public function testAddFrameReadsCodeFromLongFile(): void
         }
     }
 
+    public function testAddFrameUsesOptionsContext(): void
+    {
+        $this->options->setContextLines(2);
```

Comment:
Can you add a data provider to this test so we can see it work with `0` and two other different values?

Reply:
Please review the latest changes - I moved the redunant tests that @ste93cry mentioned into the dataprovider and created a few additional ones (zero context, context near end and beginning of file).

Reply:
Still some more minor fixes, the rest looks good!